src/components/structures/RoomStatusBar.tsx

```diff
@@ -276,6 +336,94 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         </>;
     }
 
+    private getRefreshTimelineContent(): JSX.Element {
+        let buttonRow = <>
+            <AccessibleButton
+                onClick={this.onRefreshTimelineClick}
+                className="mx_RoomStatusBar_refreshTimelineBtn"
+                data-cy="refresh-timeline-button"
+            >
+                { _t("Refresh timeline") }
+            </AccessibleButton>
+        </>;
+        if (this.state.isRefreshing) {
+            buttonRow = <>
+                <InlineSpinner w={20} h={20} />
+                { /* span for css */ }
+                <span>{ _t("Refreshing") }</span>
```

Comment:
Copied from the unsent messages status bar, https://github.com/matrix-org/matrix-react-sdk/blob/f3b762c1a807d376a52a794cf6e7e9fd26de8bf9/src/components/structures/RoomStatusBar.tsx#L248-L252

As mentioned in https://github.com/matrix-org/matrix-react-sdk/pull/8354#discussion_r882228796, the styles here could use some love in general but we can iterate that in a separate PR.